### PR TITLE
Update seed database to 2.1.0

### PR DESCRIPTION
**Pull request by oplantalech**
Seed database files and documentation have been updated to version 2.1.0, and they are suitable for cBioPortal 1.5.0 without having to execute the migration script. @pieterlukasse

### CODE REVIEW

**Comment by oplantalech** [comment]
I have added an extra documentation file with a brief explanation on how to update the seed database.

File: seedDB/Update-Seed-Database.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+To update the seed database to a recent version you should follow these steps:
+
+1- Set up a new instance of the cBioPortal database, by using the most recent available seed database: [more information](https://github.com/cbioportal/datahub/blob/master/seedDB/README.md)
+
+2- Run the migration script: [more information](https://github.com/cBioPortal/cbioportal/blob/master/docs/Updating-your-cBioPortal-installation.md#running-the-migration-script)
+
+3- Move to the folder where you want to save the seed files and generate them. To do that, use the following commands (assuming that the database is running on port 8306):
+
+```shell
+mysqldump -u cbio -pP@ssword1 -P 8306 --host 127.0.0.1 --ignore-table cbioportal.pdb_uniprot_alignment --ignore-table cbioportal.pdb_uniprot_residue_mapping cbioportal > seed-cbioportal_no-pdb_hg19.sql
```
Comment (sandertan) [inline seedDB/Update-Seed-Database.md]: Please add `--no-create-info`. We don't want to include the schema in this file. When importing, we get the schema from cgds.sql

Comment (sandertan) [inline seedDB/Update-Seed-Database.md]: Also, please include `--complete-insert` (without argument). This is for safety, that column names are included in the insert statement.

File: seedDB/Update-Seed-Database.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+mysqldump -u cbio -pP@ssword1 -P 8306 --host 127.0.0.1 --no-create-info cbioportal pdb_uniprot_alignment pdb_uniprot_residue_mapping > seed-cbioportal_only-pdb.sql
```
Comment (sandertan) [inline seedDB/Update-Seed-Database.md]: Same as comment above, please include `--complete-insert`.

**Comment by sandertan** [comment]
Updated the seed database to schema v2.0.1 to be compatible with cBioPortal 1.5.0. Includes the fix from https://github.com/cBioPortal/cbioportal/pull/2210 which was necessary for this PR.

**Comment by oplantalech** [comment]
@sandertan I've tested your update to 2.0.1. Everything looks good, great work!

**Comment by sandertan** [comment]
I updated the filenames and documentation to match the database version (2.1.0) of the released cBioPortal 1.5.0, to avoid confusion.

**Comment by Sjoerd-van-Hagen** [comment]
@sheridancbio 

Hi Robert,

Oleguer listed you as a suggested reviewer, are you the right person for this and do you have time to review this PR, or should we ask someone else at MSK. If so, who would be the best option?

Thanks,

Sjoerd.

File: seedDB/Update-Seed-Database.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Update cBioPortal seed database files 
+To update the seed database files to a recent version you should follow these steps:
```
Comment (pieterlukasse) [inline seedDB/Update-Seed-Database.md]: maybe to make it very clear that this is documentation for developers intending to update the files in github, start with something like:
"For developers: to update the seed database files in this github repository to a new version you should follow these steps:"

**Comment by oplantalech** [comment]
@pieterlukasse I changed a bit the writing in the documentation files. Hopefully it is more clear that the updating is related to the files stored in datahub. However, I thought that when talking about "seed database files" it was clear enough that those are the ones stored in datahub.

File: seedDB/Update-Seed-Database.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# Update cBioPortal seed database files stored in datahub 
+This documentation file is addressed to developers. To update the seed database files to a recent version you should follow these steps:
+
+1. Start a new instance of the cBioPortal database with the previous seed database ([more information](README.md)).
+
+2. Run the migration script from a branch that includes the new database schema ([more information](https://github.com/cBioPortal/cbioportal/blob/master/docs/Updating-your-cBioPortal-installation.md#running-the-migration-script)).
+
+3. Move to the folder where you want to save the seed files. Use the following commands (assuming that the database is running on port 8306) to generate the new seed files. Please specify the new schema version in the file name (i.e. `v2.1.0`).
```
Comment (pieterlukasse) [inline seedDB/Update-Seed-Database.md]: not "i.e. but e.g."

File: seedDB/README.md [hunk 5]
```diff
@@ -1,17 +1,20 @@
...
 
 You can download the files by using the links below:
 
-- **Schema 1.3.0**: [SQL file with create table statements for portal release 1.3.1](https://raw.githubusercontent.com/cBioPortal/cbioportal/v1.3.1/core/src/main/resources/db/cgds.sql) 
-- **Seed data, part1**: [cbioportal-seed SQL (.gz) file - part1 (no pdb_ tables)](https://github.com/cbioportal/datahub/raw/b69c86803c40d543080bf31a645721d06c82d08d/seedDB/seed-cbioportal_no-pdb_hg19.sql.gz)
-- **Seed data, part2 (optional)** [cbioportal-seed SQL (.gz) file - part2 (only pdb_ tables)](https://github.com/cbioportal/datahub/raw/b69c86803c40d543080bf31a645721d06c82d08d/seedDB/seed-cbioportal_only-pdb.sql.gz)
+- **Schema 2.1.0**: [SQL file with create table statements for portal release 1.5.0](https://raw.githubusercontent.com/cBioPortal/cbioportal/v1.5.0/db-scripts/src/main/resources/cgds.sql) 
+- **Seed data, part1**: [cbioportal-seed SQL (.gz) file - part1 (no pdb_ tables)](seed-cbioportal_hg19_v2.1.0.sql.gz)
+- **Seed data, part2 (optional)** [cbioportal-seed SQL (.gz) file - part2 (only pdb_ tables)](seed-cbioportal_hg19_v2.1.0_only-pdb.sql.gz)
```
Comment (pieterlukasse) [inline seedDB/README.md]: by introducing version in the file name, other parts of documentation also need to change:
- the docker documentation in https://github.com/thehyve/cbioportal-docker/blob/master/README.md. Please make a PR there to reflect the fact that version nr part of the file name now.
- make this version nr more explicit in your new doc `Update-Seed-Database.md`

**Comment by pieterlukasse** [comment]
some more comments regarding versioning now

**Comment by oplantalech** [comment]
@pieterlukasse I've improved the `Update-Seed-Database.md` file, and I've created a new PR in cbioportal-docker to update the documentation with the new seed db names (thehyve/cbioportal-docker#13).

**Comment by pieterlukasse** [comment]
@oplantalech good job. I also tested it and did some manual review against previous seed version (1.3.0) and all looks good :+1: 

Here are some of the things I tested (except pdb tables):

- Installed seed DB 1.3.0 from `master`
- Installed seed DB of this PR
- Got a list of tables and exact counts from 
```sql
SELECT CONCAT(
    'SELECT "', 
    table_name, 
    '" AS table_name, COUNT(*) AS exact_row_count FROM `', 
    table_schema,
    '`.`',
    table_name, 
    '` UNION '
) 
FROM INFORMATION_SCHEMA.TABLES 
WHERE table_schema = 'cbioportal';
```
- Compared the changes between both versions and manually reviewed against `migration.sql`
image

As I said, all looks good :+1: I found no errors in my review.

File: seedDB/README.md [hunk 6]
```diff
@@ -1,17 +1,20 @@
...
-- DB schema version **1.3.0**
+These files are MySQL database dump files for seeding a new instance of the cBioPortal DB. They contain all the necessary background data for a properly functioning cBioPortal website, including cancer types, gene, uniprot-mappings, drug, pdb and network data. Data is compatible with:
+
+- DB schema version **2.1.0**
 
 Which is suitable for:
 
-- release version **1.3.1** of cBioPortal.  
-- release version **> 1.3.1** of cBioPortal (:warning: with possible migration script step). 
+- release version **1.5.0** of cBioPortal.
```
Comment (pieterlukasse) [inline seedDB/README.md]: we can state **1.5.1** here. I just checked, and 1.5.1 does *not* introduce updates to DB schema.

Comment (pieterlukasse) [inline seedDB/README.md]: I also tested loading a public study using SKCM study (https://github.com/cBioPortal/datahub/blob/master/public/skcm_tcga.tar.gz). It loaded fine :+1: 

Some screenshots:
image

image

image

**Comment by pieterlukasse** [comment]
I also tested loading a public study using SKCM study (https://github.com/cBioPortal/datahub/blob/master/public/skcm_tcga.tar.gz). It loaded fine :+1: 

Some screenshots:
image

image

image

**Pull request merged**